src/chainstate/coordinator/mod.rs

```diff
@@ -589,6 +598,43 @@ impl<'a, T: BlockEventDispatcher, N: CoordinatorNotices, U: RewardSetProvider>
                     increment_stx_blocks_processed_counter();
                     let block_hash = block_receipt.header.anchored_header.block_hash();
 
+                    let mut attachments_instances = HashSet::new();
+                    let atlas_config = AtlasConfig::default();
+                    for receipt in block_receipt.tx_receipts.iter() {
+                        if let TransactionPayload::ContractCall(ref contract_call) =
+                            receipt.transaction.payload
+                        {
+                            let contract_id = contract_call.to_clarity_contract_id();
+                            if atlas_config.contracts.contains(&contract_id) {
+                                for event in receipt.events.iter() {
+                                    if let StacksTransactionEvent::SmartContractEvent(
+                                        ref event_data,
+                                    ) = event
+                                    {
+                                        let res = AttachmentInstance::try_new_from_value(
+                                            &event_data.value,
+                                            &contract_id,
+                                            &block_receipt.header.consensus_hash,
+                                            block_receipt.header.anchored_header.block_hash(),
+                                            block_receipt.header.block_height,
+                                        );
+                                        if let Ok(attachment_instance) = res {
+                                            attachments_instances.insert(attachment_instance);
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                    if !attachments_instances.is_empty() {
+                        match self.attachments_tx.send(attachments_instances) {
+                            Ok(_) => {}
+                            Err(e) => {
+                                error!("Error dispatching attachments {}", e);
```

Comment:
Should this code re-try sending the attachment if the send fails?  Or, is it okay if this code silently drops the attachment instances?

Reply:
This code is sending attachment instances over a channel. If for some reason, this channel broke, then it's probably broken forever, and re-trying later would lead to the same result. The current strategy is to log an error and move forward - I think we agreed on the fact that Atlas should not be introducing any perturbation in consensus / chain progress. 
Happy to revisit.

Reply:
> If for some reason, this channel broke, then it's probably broken forever, and re-trying later would lead to the same result

If the node can't recover, then I think the node should panic.  Otherwise, this node (and others) may keep running for a long time while silently dropping zone files, which is a far worse network-wide problem from which there is no recovery.

Reply:
I see. I've tried to stick to the policy "networking code should not panic" but yeah this site in on the other side.
Addressed with https://github.com/blockstack/stacks-blockchain/pull/2013/commits/72b8414b5d3aacb152278d74d1fcc5370df73552